File: php/factoryPattern/Cookie.php
<?php namespace php\factoryPattern;
/*
|===========================================================
|
|-----------------------------------------------------------
|	The Cookie is the 'product' in this pattern
|-----------------------------------------------------------
|
|===========================================================
|
*/
class Cookie {
	
	/**
	 * Name of cookie
	 * 
	 * @var string $name
	 */
	protected $name;
		
	/**
	 * Name of cookie topping
	 * 
	 * @var string $topping
	 */
	protected $topping;
		
	/**
	 * Name of flavour
	 * 
	 * @var string $flavour
	 */
	protected $flavour;
	
	/**
	 * Display name of flavour
	 * 
	 * @return void
	 */
	public function mix() {
		echo 'mix in the ' . $this->flavour . '<br>';
	}
		
	/**
	 * Display name of topping
	 * 
	 * @return void
	 */
	public function bake() {
		echo 'Baking and sprinkle ' . $this->topping . ' on top<br>';
	}
		
	/**
	 * Display name of cookie
	 * 
	 * @return void
	 */
	public function box() {
		echo 'Your ' . $this->name . ' cookie is ready<br>';
	}
}

File: java/factoryPattern/Cookie.java
package factoryPattern;
/*
|===========================================================
|
|-----------------------------------------------------------
|	The Cookie is the 'product' in this pattern
|-----------------------------------------------------------
|
|===========================================================
|
*/
public class Cookie {
	protected String flavour;
	protected String name;
	protected String topping;
	public Cookie() {

	}

	public void mix() {
		System.out.println("Mixing in the " + flavour);
	}

	public void bake() {
		System.out.println("baking with " + topping + " on top");
	}

	public void box() {
		System.out.println("Your " + name + " is ready now.");
	}
}
